tests/test_git.py

```diff
@@ -817,6 +817,12 @@ def test_gitpath_init(self, mock_expanduser):
         cmd = GitCommand(*args)
         self.assertEqual(cmd.parsed_args.gitpath, '/tmp/gitpath')
 
+        args = ['http://example.com/',
+                '--base-path', '/tmp/basepath']
+
+        cmd = GitCommand(*args)
+        self.assertEqual(cmd.parsed_args.gitpath, '/tmp/basepath/http://example.com/' + '-git')
```

Comment:
Is there any problem with this other form?
```suggestion
        self.assertEqual(cmd.parsed_args.gitpath, '/tmp/basepath/http://example.com/-git')
```